classpath/javax/net/ssl/SSLEngine.java

```diff
@@ -0,0 +1,100 @@
+package javax.net.ssl;
+
+import java.lang.Object;
+import java.nio.ByteBuffer;
+import java.nio.ReadOnlyBufferException;
+
+public abstract class SSLEngine extends Object {
+    private final String peerHost;
+    private final int peerPort;
+
+    protected SSLEngine() {
+        peerHost = null;
+        peerPort = -1;
+    }
+    protected SSLEngine(String peerHost, int peerPort){
+        this.peerHost = peerHost;
+        this.peerPort = peerPort;
+    }
+    public String getPeerHost(){
+        return peerHost;
+    }
+    public int getPeerPort(){
+        return peerPort;
+    }
+    public SSLEngineResult wrap(ByteBuffer src, ByteBuffer dst) throws SSLException {
+        return wrap(new ByteBuffer [] {src}, 0, 1, dst);
+    }
+    public SSLEngineResult wrap(ByteBuffer[] srcs, ByteBuffer dst) throws SSLException {
+        if (srcs == null){
+            throw new IllegalArgumentException ("srcs == null");
+        }
+        return wrap(srcs, 0, srcs.length, dst);
+    }
+    public abstract SSLEngineResult wrap(ByteBuffer[] srcs, int offset, int length, ByteBuffer dst) throws SSLException;
```

Comment:
in general its a good idea to put all the implemented methods together and as well as the abstract ones.